samza-core/src/main/java/org/apache/samza/clustermanager/ClusterBasedJobCoordinator.java

```diff
@@ -136,6 +141,11 @@
    */
   private final StreamPartitionCountMonitor partitionMonitor;
 
+  /**
+   * The input stream regex monitor
+   */
+  private final Optional<InputStreamRegexMonitor> inputStreamRegexMonitor;
```

Comment:
Unrelated to your PR, do you mind also using an Optional for the StreamPartitionCountMonitor instance?

Reply:
done